src/seedsigner/views/view.py

```diff
@@ -253,19 +253,26 @@ def run(self):
 
     class DoResetThread(BaseThread):
         def run(self):
+            import os
+            import sys
             import time
             from subprocess import call
 
             # Give the screen just enough time to display the reset message before
             # exiting.
             time.sleep(0.25)
 
-            # Kill the SeedSigner process; Running the process again.
             # `.*` is a wildcard to detect either `python`` or `python3`.
+            kill_cmd = "kill $(pidof python*)"
             if Settings.HOSTNAME == Settings.SEEDSIGNER_OS:
-                call("kill $(pidof python*) & python /opt/src/main.py", shell=True)
+                python_exec = "python" # Use system python on SeedSigner OS
+                script_path = "/opt/src/main.py" # Path to SeedSigner main script
             else:
-                call("kill $(ps aux | grep '[p]ython.*main.py' | awk '{print $2}')", shell=True)
+                python_exec = sys.executable  # Current Python interpreter path
+                script_path = os.path.abspath(sys.argv[0]) # Absolute path to current script
```

Comment:
See: https://docs.python.org/3/library/sys.html#sys.argv

"it is operating system dependent whether this is a full pathname or not"

So I think `os.path.abspath()` will fail(?) if `sys.argv[0]` just returns "main.py".